build.gradle

```diff
@@ -590,6 +607,9 @@ dependencies {
     compileOnly "org.opensearch:opensearch-job-scheduler-spi:${job_scheduler_version}"
     implementation "org.opensearch:common-utils:${common_utils_version}"
     implementation "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
+    implementation 'software.amazon.randomcutforest:randomcutforest-parkservices:3.0-rc2'
```

Comment:
Could you list the tests you have done to upgrade to rc2? What I can think of:
* real time single stream and HCAD (including create/start detector, see detector emit results; then stop cluster and restart it to see if results continue to show)
* historical single stream and HCAD
* backward compatible tests: v1 to v3 model, v2 to v3 model: v1 model in checkpoint should not be overridden, and we add v2 and v3 model in checkpoint if necessary.
* Does memory size formula still hold (Check MemoryTracker)?

Reply:
I tested single stream real time and historical, I;ll do the same for HCAD right now. BWC test were the test added based on Yaliang's code to check deserialize model from 1.3. And converting test to v1 to v3, I didn't add v2 to v3.

Reply:
The unit tests for bwc is not enough. It just checks the conversion has no exception. We will do some manual testing suggested above.

Reply:
I manually tested that after starting both single and HCAD (both real time and historical) with current RCF and getting results, I can switch to RCF 3.0-rc2 and there is no issues, detectors continue to run the same

Reply:
After letting the detector run for longer I do get one issue, `Failed to generate checkpoint, java.lang.IllegalStateException: There is discepancy in indices`